src/config.js

```diff
@@ -0,0 +1,391 @@
+   * @param {object[]} options.extendedPublicKeys - array of extended pub key objects
+   * @param {string} [options.extendedPublicKeys.xfp] - master fingerprint
+   * @param {string} options.extendedPublicKeys.xpub - valid xpub
+   * @param {string} options.extendedPublicKeys.xpub - valid xpub
+   * @param {string} [options.extendedPublicKeys.bip32Path] - bip32path corresponding to xpub
+   * @param {string} [options.extendedPublicKeys.name] - name of key
+   * @param {string} [options.network='mainnet'] - network corresponding to keys
+   * @param {string} [options.client] - private or public, really only valid for caravan
+   * @returns {MultisigWalletConfig} new config instance
+   */
+  constructor({ 
+    uuid, 
+    name, 
+    quorum, 
+    addressType, 
+    extendedPublicKeys,
+    startingAddressIndex,
+    network,
+    client,
+  }) {
+    
+    this.supportedNetworks = [MAINNET, TESTNET];
+    this.supportedClients = ['public', 'private'];
+
+    assert(uuid || name, 'Name or UUID required to create new MultisigWalletConfig')
+    this.name = uuid || name;
+
+    if (network) {
+      assert(
+        this.supportedNetworks.includes(network), 
+        `Network ${network} not supported.`
+      )
+      this.network = network;
+    } else {
+      this.network = MAINNET;
+    }
+    
+    if (client) {
+      assert(
+        this.supportedClients.includes(client),
+        `Client type "${client}" not supported`
+      )
+      this.client = client;
+    }
+
+    assert(
+      quorum && quorum.requiredSigners && quorum.totalSigners,
+      "Configuration file needs quorum.requiredSigners and quorum.totalSigners."
+    );
+    this.requiredSigners = quorum.requiredSigners;
+    this.totalSigners = quorum.totalSigners;
+    
+    assert(typeof addressType === 'string', "Configuration file needs addressType.")
+    this.addressType = addressType;
+
+    assert(
+      extendedPublicKeys && 
+      Array.isArray(extendedPublicKeys),
+      "Configuration file needs array of extendedPublicKeys."
+    ) 
+    this.extendedPublicKeys = extendedPublicKeys;
+    this.validateExtendedPublicKeys(false);
+
+    if (startingAddressIndex) {
+      if (typeof startingAddressIndex !== 'number') {
+        throw new TypeError('Expected int for startingAddressIndex');
+      }
+      this.startingAddressIndex = startingAddressIndex;
+    }
+  }
+
+  /**
+   * @description instantiate a new wallet config object from a JSON
+   * @param {string} jsonString - json string to convert.
+   * @returns {MultisigWalletConfig} returns a new instance of a MultisigWalletConfig
+   */
+  static fromJSON(jsonString) {
+    if (typeof jsonString !== 'string') throw new TypeError('Must pass a valid JSON string');
+    let options;
+
+    try {
+      options = JSON.parse(jsonString);
+    } catch (error) {
+      throw new Error("Unable to parse JSON.");
+    }
+    
+    // TODO: how do we handle caravan configs where the extended public keys
```

Comment:
Yeah, exactly. Or really any JSON config we can get. Presumably not all config setups have or require xfps since you can configure a multisig wallet without it.

Reply:
@humanumbrella any further comment on this or good to just leave as a TODO?

Reply:
My first thought was just start counting -- previous to `3.2.1` you could set em all to `0x00000000` but now that won't work - they need to be unique. So your suggestion of using the fingerprint that's in the xpub (even though it's not the root fingerprint) would probably be fine. 

If none of the xpubs have xfps - we should not do this though, because none of them will be able to sign (unless just happen to be at depth 1 and the parent fingerprint of the xpub *is* the root fingerprint.